packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/middleware/assertRateLimiter.test.ts

```diff
@@ -0,0 +1,70 @@
+import { afterAll, beforeAll, beforeEach, describe, expect, it } from "vitest";
+import { RateLimiter } from "./rateLimiter";
+import { config } from "../config";
+import { getRedisConnection } from "../redis";
+
+beforeEach(async () => {
+    // run with npm run test with audius-compose up
+    config.redisUrl = "redis://0.0.0.0:6379/00"
+    const redis = await getRedisConnection()
+    let cursor = '0';
+    do {
+        for await (const key of redis.scanIterator()) {
+            if (key.startsWith('listens-rate-limit')) {
+                await redis.del(key)
+            }
+        }
+    } while (cursor !== '0')
+})
+
+describe('Rate Limiter', function () {
+    describe('Rate Limiter Rate Limits', function () {
+        it('disallows when one limit is reached', async function () {
+            const userIP = "1.2.3.4"
+
+            const hourLimitLimiter = new RateLimiter("listens-rate-limit-hour", 1, 2, 2)
+            const checkHour = await hourLimitLimiter.checkLimit(userIP)
+            expect(checkHour.allowed).toBe(false)
+            expect(checkHour.hourLimitReached).toBe(true)
```

Comment:
Just out of curiosity, what was the reason for returning all of these as individual booleans? Are we using them for some sort of logic? Or was this just to make testing easier so we could see _which_ limit was hit?

Reply:
to see which limit was hit in the logs for debugging purposes